### ISSUE TITLE
Unexpected token P in JSON at position 0 azure developer CLI extension vscode

### ISSUE THREAD

**Issue by amit-pazcare** [Azure/azure-dev#2072]
Does this occur consistently? 
Repro steps:

1.
2.

Action: azure-dev.views.workspace.environment.resolve
Error type: 1
Error Message: Process 'azd env list --cwd "/Users/redacted:username/Deskto...' exited with code 1
Error: {"type":"consoleMessage","timestamp":"redacted:id02T23redacted:id+redacted:id","data":{"messhttps://docs.github.com/en/early-access/issues/about-tasklistsage":"listing environments: listing entries: open /Users/redacted:username/Desktop/msdocs-flask-postgresql-sample-app/.azure: no such file or directory\n"}}

Version: 0.4.2
OS: darwin
OS Release: 22.1.0
Product: Visual Studio Code
Product Version: 1.76.0
Language: en

<details>
<summary>Call Stack</summary>

```
ChildProcess.<anonymous> extension.js:2:919933
```

</details>

**Comment by ellismg** [comment]
So, clearly bad of the CLI to give back a string instead of JSON in this error case, but from the error message, it sounds like the `.azure` folder didn't exist, which seems like an issue in itself?

**Comment by bwateratmsft** [comment]
Very similar to #2011

**Comment by rajeshkamal5050** [comment]
@weikanglim does your fix above handle this error scenario? so VSCode gets a proper error json?

**Comment by weikanglim** [comment]
@rajeshkamal5050 The underlying "listing environments" implementation was fixed, so it's no longer an issue if the user doesn't have an `.azure` folder, but the output isn't fixed

**Comment by weikanglim** [comment]
@bwateratmsft Based on the info above, is there something we need to do in the CLI? I can't tell if the output is the raw output that VSCode got from the CLI. 

From the error message "Unexpected token P at position 0", could the token be referring to the "Process" in 
`Error Message: Process 'azd env list --cwd "/Users/redacted:username/Deskto...' exited with code 1`?

**Comment by rajeshkamal5050** [comment]
Marking this as fixed as part of #2296 

```
acme@devdiv azd-demo % azd env list --output json
[]

acme@devdiv azd-demo % azd show --output json
{
  "name": "todo-python-mongo",
  "services": {
    "api": {
      "project": {
        "path": "/Users/rajeshkamal/azd-demo/src/api",
        "language": "python"
      }
    },
    "web": {
      "project": {
        "path": "/Users/rajeshkamal/azd-demo/src/web",
        "language": "node"
      }
    }
  }
}
```
